lib/core/src/client/preview/start.ts

```diff
@@ -68,6 +68,16 @@ export default function start(
     const selectionSpecifier = getSelectionSpecifierFromPath();
     if (selectionSpecifier) {
       storyStore.setSelectionSpecifier(selectionSpecifier);
+
+      const { args, storySpecifier } = selectionSpecifier;
+      if (args && typeof storySpecifier === 'string') {
+        channel.on(Events.SET_STORIES, () => {
+          channel.emit(Events.UPDATE_STORY_ARGS, {
+            storyId: storySpecifier,
+            updatedArgs: args,
+          });
+        });
+      }
```

Comment:
Is this to update the args on the first story? Shouldn't `setSelectionSpecifier({args})` do this already?

It seems incorrect that:

 (a) the store itself isn't responsible for setting the initial arg value in this case
 (b) we are going to emit `UPDATE_STORY_ARGS`/`STORY_ARGS_UPDATED` when technically that hasn't happened.

Reply:
I would have guessed `store.finishConfiguring` should call `store.selectSelection` with the args (just like it does with the `viewMode`) and things would flow fairly naturally from there. Is there a reason it didn't work that way?